src/js/tech/ChromecastTech.js

```diff
@@ -230,52 +233,92 @@ ChromecastTech = {
           customData = this._requestCustomData(source),
           textTrackJsonTracks = this.videojsPlayer.textTracksJson_,
           request,
+          castSessionObj,
           i;
 
-      mediaInfo.entity = loadSource.entity;
-      mediaInfo.contentUrl = loadSource.src;
-      mediaInfo.contentType = loadSource.type;
-
       this.trigger('waiting');
       this._clearSessionTimeout();
 
-      mediaInfo.metadata = new chrome.cast.media.GenericMediaMetadata();
-      mediaInfo.metadata.metadataType = chrome.cast.media.MetadataType.GENERIC;
-      mediaInfo.metadata.title = title;
-      mediaInfo.metadata.subtitle = subtitle;
-      mediaInfo.streamType = this.videojsPlayer.liveTracker && this.videojsPlayer.liveTracker.isLive()
-         ? chrome.cast.media.StreamType.LIVE
-         : chrome.cast.media.StreamType.BUFFERED;
-      mediaInfo.tracks = [];
-      mediaInfo.activeTrackIds = [];
-
-      for (i = 0; i < textTrackJsonTracks.length; i++) {
-         mediaInfo.tracks.push(this.generateTrack(textTrackJsonTracks[i], i));
-         if (textTrackJsonTracks[i].mode === 'showing') {
-            mediaInfo.activeTrackIds.push(i);
+      // if more then one source was load, load queue
+      if (loadSource.sources) {
+         this._queue = loadSource.sources;
+         const queueMediaInfo = loadSource.sources.map((queueItem) => {
+            const mediaInfoItem = new chrome.cast.media.MediaInfo(queueItem.src, queueItem.type);
+
+            mediaInfoItem.entity = queueItem.entity;
+
+            mediaInfoItem.contentUrl = queueItem.src;
+            mediaInfoItem.metadata = new chrome.cast.media.GenericMediaMetadata();
+            mediaInfoItem.metadata.title = queueItem.title;
+            mediaInfoItem.duration = queueItem.duration;
+            mediaInfoItem.metadata.subtitle = queueItem.subtitle;
+            mediaInfoItem.streamType = this.videojsPlayer.liveTracker && this.videojsPlayer.liveTracker.isLive()
+               ? chrome.cast.media.StreamType.LIVE
+               : chrome.cast.media.StreamType.BUFFERED;
+            mediaInfoItem.tracks = [];
+            mediaInfoItem.activeTrackIds = [];
+
+            return new chrome.cast.media.QueueItem(mediaInfoItem);
+         });
```

Comment:
Can (most of) this callback be reused in both `if (loadSource.sources)` branches?

Reply:
I moved as much as possible of the code outside of conditional statements.